apps/dg/controllers/document_controller.js

```diff
@@ -553,12 +553,115 @@ DG.DocumentController = SC.Object.extend(
       restoreComponentControllersAndViews: function () {
         var components = this.get('components');
         if (components) {
+          if (DG.KEEP_IN_BOUNDS_PREF) {
+            this.set('scaleFactor', 1);
+            this.set('scaleBoundsX', 0);
+            this.set('scaleBoundsY', 0);
+            this.computeScaleBounds();
+            if (Object.keys(components).length) {
```

Comment:
In this case, components is actually an object with properties based on the ids so a simple `.length` returns undefined.  Since `Object.keys` returns an array of the properties we can then get the length to see if the object has any key/value pairs.

Reply:
Right you are!